ls1intum/Artemis
# `General`: Fix an issue with an unintentional log out during unsaved changes prevented correctly

### Checklist
#### General



#### Client

### Motivation and Context


Fixes #5440 issue.

### Description

Previously, during exam or exercise if user tries to logout, the logout logic is being executed then guard is fired. In this PR i changed it to first navigate to `/` route in order to check if the Pending Changes guard will fire, and only if user confirms to logout then it will execute the logout logic.

### Steps for Testing

Prerequisites:
- 1 Admin
- 1 Students
- 1 Modeling Exercise

1. Log in to Artemis as a admin
2. Navigate to Courses management and create a new course with modeling exercise 
3. Now log in to Artemis as a student
4. Navigate to Courses
5. Select the course and navigate to exercises tab
6. Open modeling editor
7. Make some changes by adding some diagrams
8. Try to logout while there is unsaved changes 
9. A confirmation popup should show up, press OK to logout or cancel to continue

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage



E2E Cypress test added in `src/test/cypress/integration/Logout.spec.ts`
